### PR TITLE
Add waffle-e2e

### PR DESCRIPTION
**Pull request by Jozwiaczek**

### CODE REVIEW

File: waffle-e2e/buildTestContracts.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const compilerVersion = '0.5.15';
+
+  await compileAndSave({
+    sourceDirectory,
+    outputDirectory,
+    compilerVersion
+  });
+};
+
+buildExampleContracts();
```
Comment (sz-piotr) [inline waffle-e2e/buildTestContracts.ts:15]: Actually you can just remove this function and just have this called directly
```ts
compileAndSave({
  sourceDirectory: './src',
  outputDirectory: './build',
  compilerVersion: '0.5.15'
})
```

File: waffle-e2e/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  "license": "MIT",
+  "scripts": {
+    "build": "ts-node buildTestContracts",
+    "test:nobuild": "export NODE_ENV=test && mocha",
+    "test": "yarn build && yarn test:nobuild",
+    "lint": "eslint '{src,test}/**/*.ts'",
+    "lint:fix": "eslint --fix '{src,test}/**/*.ts'"
+  },
+  "dependencies": {
+    "@ethereum-waffle/compiler": "^3.0.0-beta.1"
```
Comment (sz-piotr) [inline waffle-e2e/package.json:14]: This project should probably only have dev dependencies

Comment (Jozwiaczek) [reply waffle-e2e/package.json:14]: I left it here because of linter warning in buildTestContracts

Comment (sz-piotr) [reply waffle-e2e/package.json:14]: add a linter comment in the script then to disable the error

Comment (sz-piotr) [reply waffle-e2e/package.json:14]: The linter is configured so that only tests can use dependencies from devDependencies

**Pull request merged**